src/ceph-volume/ceph_volume/devices/lvm/strategies/strategies.py

```diff
@@ -2,13 +2,21 @@
 
 class Strategy(object):
 
-    def __init__(self, devices, args):
+    def __init__(self, data_devs, db_devs, wal_devs, args):
         self.args = args
         self.osds_per_device = args.osds_per_device
-        self.devices = devices
-        self.hdds = [device for device in devices if device.sys_api['rotational'] == '1']
-        self.ssds = [device for device in devices if device.sys_api['rotational'] == '0']
-        self.computed = {'osds': [], 'vgs': [], 'filtered_devices': args.filtered_devices}
+        self.devices = data_devs + wal_devs + db_devs
+        self.data_devs = data_devs
+        self.db_devs = db_devs
+        self.wal_devs = wal_devs
+        self.computed = {'osds': [], 'vgs': []}
+
+    @staticmethod
+    def split_devices_rotational(devices):
+        data_devs = [device for device in devices if device.sys_api['rotational'] == '1']
+        db_devs = [device for device in devices if device.sys_api['rotational'] == '0']
+        return data_devs, db_devs
```

Comment:
what happens here for wal devs? why would this omit them

Reply:
See comment above.